Blaze: ensure the Dashboard menu is only inserted once.

jeherve opened the pull request:
## Proposed changes:

This fixes issues on WoA sites and WordPress.com Simple sites, where when the new dashboard is enabled, the navigation menu (under Tools > Advertising) is added by the package, while the old Calypso link is also added, via the masterbar.

To solve this problem, we rely on the package, and only the package, to register the menu. **We do point WoA and simple site owners to the Calypso advertising page (and not the wp-admin advertising page) at all times, since they have nav-unification on and are logged in to WordPress. com at all times.**

This PR also changes the order of the new menu added by the package, to ensure it's always added at the top of the Tools menu.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* p1687955577164529/1687894794.905929-slack-C053FAERCDP

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

You'll want to test this on all three site types (WoA, Simple, self-hosted).

- On WoA sites, you should see a Tools > Advertising menu item pointing you to Calypso at all times.
- On Simple sites, you should see a Tools > Advertising menu item pointing you to Calypso at all times.
- On self-hosted sites, you should not see any Tools > Advertising menu item by default.
    - If you enable the Blaze feature under Jetpack > Settings > Traffic, you should still see no new menu item.
    - If you enable the Blaze dashboard with `add_filter( 'jetpack_blaze_dashboard_enable', '__return_true' );`, you should see a new menu item, pointing you to the wp-admin advertising page.

jeherve committed:
Blaze: ensure the Dashboard menu is only inserted once.

This fixes issues on WoA sites and WordPress.com Simple sites, where when the new dashboard is enabled, it is added by the package while the old Calypso link is also added in the masterbar.

jeherve committed:
Remove nav-unification handling, let the package register things

jeherve committed:
Always prefer calypos view in nav unification

jeherve committed:
Remove Blaze from test now that menu is handled in package

jeherve committed:
Move more of the logic into the package to simplify things

sbarbosa commented on projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php:
```diff
@@ -381,10 +380,6 @@ public function add_options_menu() {
 		// performance settings already have a link to Page Optimize settings page.
 		$this->hide_submenu_page( 'options-general.php', 'page-optimize' );
-
-		if ( Blaze::should_initialize() ) {
```
I checked and this code is now inside the Blaze package. This means that the Advertising menu will appear only if the Blaze module is enabled. Right?

I think it is already like this right now, you are just moving this code to the blaze package.

jeherve replied:
That's correct, yes.

sbarbosa reviewed (dismissed):
I tested in simple/atomic and self-hosted sites, and the menu option always pointed to the correct place. (wpcom for simple/atomic, and new dashboard for self-hosted).

The code looks good 👍

sbarbosa commented:
One more question @jeherve, after we merged/deployed this change. Would it be a way to test the Blaze Dashboard on atomic sites? 
I just want to know how can we proceed with the testing after this. Thanks!

jeherve commented:
> Would it be a way to test the Blaze Dashboard on atomic sites?
> I just want to know how can we proceed with the testing after this. Thanks!

That's a very good point, and something I should have addressed in this PR. Since the menu is not registered on WordPress.com sites, you cannot and should not be able to get to it. Right now you still can, via the quick links in the posts list, or via the post-publish panel. I'll get this fixed.

jeherve committed:
Force the dashboard off on WordPress.com sites.

jeherve committed:
Reverse menu registration logic to prioritize new dash registration

We want to be able to test the new Dashboard on WoA if we want to, by supplying the jetpack_blaze_dashboard_enable filter.
